2-projects/1-todo-list/script.js

```diff
@@ -17,6 +62,55 @@ function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
   // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  const liEl = document.createElement("li");
+
+  liEl.setAttribute(
+    "class",
+    "list-group-item d-flex justify-content-between align-items-center"
+  );
+
+  const inputValue = document.querySelector("#todoInput").value;
+  const textValueFromInput = document.createTextNode(inputValue);
+
+  liEl.appendChild(textValueFromInput);
+
+  if (inputValue === "") {
+    alert("It is empty, write something.");
+  } else {
+    ulEl.appendChild(liEl);
+  }
+
+  document.querySelector("#todoInput").value = "";
+
+  // span part
+  const spanEl = document.createElement("span");
+  // class of the li
+  spanEl.setAttribute("class", "badge bg-primary rounded-pill");
+
+  //create i
+  const iElCheck = document.createElement("i");
+  const iElTrash = document.createElement("i");
+
+  // class of the i
+  iElCheck.setAttribute("class", "fa fa-check");
+  iElTrash.setAttribute("class", "fa fa-trash");
+
+  // add the element to the UL
+  ulEl.appendChild(liEl);
+  liEl.appendChild(spanEl);
+  spanEl.append(iElCheck, iElTrash);
+
+  iElCheck.addEventListener("click", function () {
+    if (liEl.style.textDecoration == "line-through") {
+      liEl.style.textDecoration = "none";
+    } else {
+      liEl.style.textDecoration = "line-through";
+    }
+  });
+
+  iElTrash.addEventListener("click", function () {
+    liEl.parentNode.removeChild(liEl);
+  });
```

Comment:
1. We see code duplicated here at functions **populateTodoList** and **addNewTodo** which might indicate something for improvement.
2. Your solution works. However, the TODO data is heavily bound to the DOM. Think of the Word Highlight example above, could you get the original paragraph easily from those <span> enclosed strings? Usually we will make sure data stays in data layer and separated from presentation layer. Its time to visit MVC model to write de- coupled GUI code.